src/network/mcpe/NetworkSession.php

```diff
@@ -428,6 +429,12 @@ public function handleDataPacket(Packet $packet, string $buffer) : void{
 
 		$timings = Timings::getReceiveDataPacketTimings($packet);
 		$timings->startTiming();
+		$ev = new DataPacketPreReceiveEvent($this, $packet->pid(), $buffer);
+		$ev->call();
+		if($ev->isCancelled()){
+			$timings->stopTiming();
```

Comment:
You can put the event call inside the `try` and avoid having this extra call (the `finally` will take care of it).